src/main/java/org/apache/hadoop/fs/ceph/CephFileSystem.java

```diff
@@ -320,9 +320,7 @@ private String selectDataPool(Path path, int repl_wanted) throws IOException {
      * something bad, so we throw any exceptions. For configured pools we
      * ignore some errors.
      */
-    int fd = ceph.__open(new Path("/"), CephMount.O_RDONLY, 0);
-    String pool_name = ceph.get_file_pool_name(fd);
-    ceph.close(fd);
+    String pool_name = ceph.get_default_data_pool_name();
```

Comment:
yes, that can solve this issue in most situation, since the temp file have the default layout
but there are two things to be considered:
1 we have to create and delete the temp file, so we have to do more things than we need. 
2 it's hard to choose a temp file name which would never conflict with existing file.